### ISSUE TITLE
Unable to deploy to latest IOS device

### ISSUE THREAD

**Issue by acrigney** [dotnet/maui#24271]
### Description

We are trying to deploy our MAUI app to ios ipad device with ios version 17.6.1
We are trying to deploy with the latest VS2022 version 17.11.0
When we run there are many missing assemblies such as Microsoft.VisualStudio.DesignTools.TapContract which are hidden assemblies.
We have these assemblies in our proj file
<ItemGroup>
	  <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="8.0.2" />
	  <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="8.0.1" />
	  <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Http" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Logging.Configuration" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Maui.Controls" Version="8.0.7" />
	  <PackageReference Include="Microsoft.Azure.Mobile.Client.SQLiteStore" Version="4.2.0" />
	  <PackageReference Include="Microsoft.Extensions.Configuration" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Logging" Version="8.0.0" />
	  <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0" />
	  <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
	  <PackageReference Include="sqlite-net-pcl" Version="1.9.172" />
	  <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.1" />
	</ItemGroup>
When we deploy with the ios simulator we get this error
System.TypeLoadException
  Message=VTable setup of type Microsoft.Maui.Controls.Editor failed
We also updated the MAUI controls and MAUI compatibility packages but we get the same errors.

### Steps to Reproduce

A dummy app build and deploys to our IOS 17.6.1 device.

### Version with bug

8.0.70 SR7

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

17.6.1

### Did you find any workaround?

no

**Comment by andrew-barlit** [comment]
Looks like I have a similar issue but I can't deploy to any iOS device because it won't even start doing that.
Initially I've discovered that my app crashes instantly when installed from TestFlight (but runs ok when installed directly onto the device or in the simulator). It was working fine about a month ago so I guess something went broken with SDK 8.0.303.
I've updated to SDK 8.0.400 and now app won't even launch on a local device (it still runs in simulator, though). Updating to 8.0.401 didn't help at all. I've tried uninstalling everything via dotnet-core-uninstall and installing 8.0.302 SDK but app won't launch anyway.
MacOS version is 14.6.1.
I have these assemblies in my project:

`<ItemGroup>
<PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.2" />
<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0" />
<PackageReference Include="SkiaSharp.Views.Maui.Controls" Version="2.88.8" />
<PackageReference Include="SkiaSharp.Views.Maui.Core" Version="2.88.8" />
<PackageReference Include="System.IO.Hashing" Version="8.0.0" />
<PackageReference Include="System.Text.Json" Version="8.0.4" />
<PackageReference Include="CommunityToolkit.Maui" Version="9.0.2" />
<PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.9" />
<PackageReference Include="Microsoft.Maui.Controls" Version="8.0.80" />
<PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="8.0.80" />
</ItemGroup>`

Here's latest logs when I'm trying to launch the app on a local device:

`Build FAILED.

/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error : clang++ exited with code 1: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error : Undefined symbols for architecture arm64: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_Interop_CallStringMethod_TArg1_REF_TArg2_UINT16_TArg3_INT_System_Buffers_SpanFunc_5_char_TArg1_REF_TArg2_UINT16_TArg3_INT_Interop_Globalization_ResultCode_TArg1_REF_TArg2_UINT16_TArg3_INT_string_", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowGreater_T_INT_T_INT_T_INT_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowGreater_T_LONG_T_LONG_T_LONG_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowGreater_T_UINT_T_UINT_T_UINT_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfGreaterThan_int_int_int_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfGreaterThan_long_long_long_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfGreaterThan_uint_uint_uint_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfLessThan_int_int_int_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfLessThan_long_long_long_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfNegativeOrZero_int_int_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfNegative_int_int_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :      (maybe you meant: _p_1276_plt_aot_instances_System_ArgumentOutOfRangeException_ThrowIfNegative_int_int_string_llvm) [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowIfNegative_long_long_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
    0 Warning(s)
    1 Error(s)`

Here's the logs from TestFlight when I was able to build the app (prior to 8.0.400 SDK):

`Incident Identifier: 2643AF5F-E1BE-46A9-BF32-161FBA7A533E
Distributor ID:      com.apple.TestFlight
Hardware Model:      iPhone12,8
Process:             EZInspection [10407]
Path:                /private/var/containers/Bundle/Application/1AE22FD7-FED1-4B57-8410-E2365B423BDB/EZInspection.app/EZInspection
Identifier:          com.ezhomeinspectionsoftware.dev
Version:             1.7.9 (570)
AppStoreTools:       15F31e
AppVariant:          1:iPhone12,8:17.4
Beta:                YES
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           com.ezhomeinspectionsoftware.dev [707]

Date/Time:           2024-08-14 11:31:33.4349 +0300
Launch Time:         2024-08-14 11:31:32.9764 +0300
OS Version:          iPhone OS 17.5.1 (21F90)
Release Type:        User
Baseband Version:    5.00.00
Report Version:      104

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Termination Reason: SIGNAL 6 Abort trap: 6
Terminating Process: EZInspection [10407]

Triggered by Thread:  0


Thread 0 name:
Thread 0 name:
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x00000001d5dfb42c __pthread_kill + 8 (:-1)
1   libsystem_pthread.dylib       	0x00000001e9b9ac0c pthread_kill + 268 (pthread.c:1721)
2   libsystem_c.dylib             	0x0000000194c9eba0 abort + 180 (abort.c:118)
3   EZInspection                  	0x0000000102888f98 0x1005fc000 + 36229016
4   EZInspection                  	0x00000001027fafe4 0x1005fc000 + 35647460
5   libsystem_platform.dylib      	0x00000001e9adee9c _sigtramp + 56 (sigtramp.c:116)
6   libsystem_pthread.dylib       	0x00000001e9b9ac0c pthread_kill + 268 (pthread.c:1721)
7   libsystem_c.dylib             	0x0000000194c9eba0 abort + 180 (abort.c:118)
8   EZInspection                  	0x00000001026c2fb4 0x1005fc000 + 34369460
9   EZInspection                  	0x00000001026b8b48 0x1005fc000 + 34327368
10  EZInspection                  	0x00000001026b8c90 0x1005fc000 + 34327696
11  EZInspection                  	0x00000001027d4200 0x1005fc000 + 35488256
12  EZInspection                  	0x0000000102718c80 0x1005fc000 + 34720896
13  EZInspection                  	0x00000001027187a0 0x1005fc000 + 34719648
14  EZInspection                  	0x000000010271a428 0x1005fc000 + 34726952
15  EZInspection                  	0x0000000102594c40 0x1005fc000 + 33131584
16  EZInspection                  	0x000000010271a650 0x1005fc000 + 34727504
17  EZInspection                  	0x0000000102717efc 0x1005fc000 + 34717436
18  EZInspection                  	0x00000001027de12c 0x1005fc000 + 35529004
19  EZInspection                  	0x00000001027d4198 0x1005fc000 + 35488152
20  EZInspection                  	0x0000000102718c80 0x1005fc000 + 34720896
21  EZInspection                  	0x00000001027187a0 0x1005fc000 + 34719648
22  EZInspection                  	0x00000001027f64fc 0x1005fc000 + 35628284
23  EZInspection                  	0x000000010271a6a8 0x1005fc000 + 34727592
24  EZInspection                  	0x0000000102717efc 0x1005fc000 + 34717436
25  EZInspection                  	0x00000001027de12c 0x1005fc000 + 35529004
26  EZInspection                  	0x00000001027d4198 0x1005fc000 + 35488152
27  EZInspection                  	0x00000001027d8ce0 0x1005fc000 + 35507424
28  EZInspection                  	0x00000001027c7bb8 0x1005fc000 + 35437496
29  EZInspection                  	0x00000001027cc044 0x1005fc000 + 35455044
30  EZInspection                  	0x0000000102778080 0x1005fc000 + 35111040
31  EZInspection                  	0x00000001027356ec 0x1005fc000 + 34838252
32  EZInspection                  	0x0000000102735470 0x1005fc000 + 34837616
33  EZInspection                  	0x0000000102714888 0x1005fc000 + 34703496
34  EZInspection                  	0x00000001027cb8e4 0x1005fc000 + 35453156
35  EZInspection                  	0x00000001027d1934 0x1005fc000 + 35477812
36  EZInspection                  	0x0000000102594148 0x1005fc000 + 33128776
37  EZInspection                  	0x0000000102595054 0x1005fc000 + 33132628
38  EZInspection                  	0x0000000102839dd4 0x1005fc000 + 35904980
39  dyld                          	0x00000001b03c1e4c start + 2240 (dyldMain.cpp:1298)

Thread 1:
0   libsystem_pthread.dylib       	0x00000001e9b940c4 start_wqthread + 0 (:-1)

Thread 2 name:
Thread 2:
0   libsystem_kernel.dylib        	0x00000001d5df61cc __psynch_cvwait + 8 (:-1)
1   libsystem_pthread.dylib       	0x00000001e9b966e4 _pthread_cond_wait + 1228 (pthread_cond.c:862)
2   EZInspection                  	0x0000000102711b5c 0x1005fc000 + 34691932
3   libsystem_pthread.dylib       	0x00000001e9b9906c _pthread_start + 136 (pthread.c:931)
4   libsystem_pthread.dylib       	0x00000001e9b940d8 thread_start + 8 (:-1)


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0xffffffff932548c7   x5: 0x0000000000000008   x6: 0x000000016f8008a0   x7: 0x000000016f7fff90
    x8: 0xbfa2cf3c721fd96e   x9: 0xbfa2cf3d9fbca7ae  x10: 0x0000000000000200  x11: 0x000000016f800000
   x12: 0x0000000000000000  x13: 0x00000001009f9c80  x14: 0x0000000000000010  x15: 0xffffffffffffffff
   x16: 0x0000000000000148  x17: 0x00000001eda37ec0  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x0000000000000103  x21: 0x00000001eda37fa0  x22: 0x0000000000000010  x23: 0x00000000000003c0
   x24: 0x0000000102d9c42c  x25: 0x0000000000000030  x26: 0x0000000000000210  x27: 0x0000000102c4a018
   x28: 0x0000000301edad80   fp: 0x000000016f8008b0   lr: 0x00000001e9b9ac0c
    sp: 0x000000016f800890   pc: 0x00000001d5dfb42c cpsr: 0x40000000
   esr: 0x56000080  Address size fault


Binary Images:
        0x1005fc000 -         0x102a87fff EZInspection arm64  <1b6cd4f8ea34396fad907b11ac6f06c1> /private/var/containers/Bundle/Application/1AE22FD7-FED1-4B57-8410-E2365B423BDB/EZInspection.app/EZInspection
        0x103a30000 -         0x103f93fff libSkiaSharp arm64  <e8dbf01e39573ad589a5df0f96660a83> /private/var/containers/Bundle/Application/1AE22FD7-FED1-4B57-8410-E2365B423BDB/EZInspection.app/Frameworks/libSkiaSharp.framework/libSkiaSharp
        0x194c29000 -         0x194ca6ff3 libsystem_c.dylib arm64e  <b122f07fa15637f3a22d64627c0c4b24> /usr/lib/system/libsystem_c.dylib
        0x1b0385000 -         0x1b0411ef7 dyld arm64e  <71846eacee653697bf7d790b6a07dcdb> /usr/lib/dyld
        0x1d5def000 -         0x1d5e28fef libsystem_kernel.dylib arm64e  <13b5134e819c3baab3004856112114cb> /usr/lib/system/libsystem_kernel.dylib
        0x1e9ade000 -         0x1e9ae4ff3 libsystem_platform.dylib arm64e  <e99607c14f1330b7bd1543c363b459c8> /usr/lib/system/libsystem_platform.dylib
        0x1e9b93000 -         0x1e9b9fff3 libsystem_pthread.dylib arm64e  <1196b6c3333d3450818ff3663484b8eb> /usr/lib/system/libsystem_pthread.dylib

EOF`

Any help or workaround would be appreciated because it's a blocker. Are there any methods to install previous version of SDK with correct workloads or whatever?

**Comment by PureWeen** [comment]
@rolfbjarne thoughts?

**Comment by rolfbjarne** [comment]
I haven't seen this before, if you could provide a test project (or even your actual project) so that we can track it down and fix it that would be great (if so, please file an issue at [here](https://github.com/xamarin/xamarin-macios/issues/new/choose)):

```
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error : clang++ exited with code 1: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error : Undefined symbols for architecture arm64: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_Interop_CallStringMethod_TArg1_REF_TArg2_UINT16_TArg3_INT_System_Buffers_SpanFunc_5_char_TArg1_REF_TArg2_UINT16_TArg3_INT_Interop_Globalization_ResultCode_TArg1_REF_TArg2_UINT16_TArg3_INT_string_", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowGreater_T_INT_T_INT_T_INT_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :       _mono_aot_aot_instancesmethod_addresses in aot-instances.dll.o [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/targets/Xamarin.Shared.Sdk.targets(1641,3): error :   "_aot_instances_System_ArgumentOutOfRangeException_ThrowGreater_T_LONG_T_LONG_T_LONG_string", referenced from: [/Users/sambridges/mobile-app-repository/EZInspection/EZInspection/EZInspection.csproj::TargetFramework=net8.0-ios]
```

> Are there any methods to install previous version of SDK with correct workloads or whatever?

@PureWeen do you guys have a rollback file for this?

**Comment by andrew-barlit** [comment]
Unfortunately, I cannot reproduce this issue on a blank project. I've even copied a couple of services and views from the main project and it could be compiled.
Current status is:

- I can launch the same project on the phone from VS for Windows connected to the same Mac, but it's much slower than working from VS Code for Mac (as usual)
- I can build (dotnet publish) the project from the Mac, but the produced ipa file is DOA (instantly crashing when installed from TestFlight, crash log is identical to the one from my previous message)
- Publish iOS option from VS for Windows never worked and still doesn't (it just never finishes, Cancel button does nothing, have to kill the process)

It looks like some dotnet packages or workloads are somehow damaged on Mac, but how could I repair them? I've already deleted everything via dotnet-core-uninstall and re-installed the latest sdk, but it still shows me a warning when installing workloads that some file with version 8.0.200 could not be found.

**Comment by rolfbjarne** [comment]
> * I can build (dotnet publish) the project from the Mac, but the produced ipa file is DOA (instantly crashing when installed from TestFlight, crash log is identical to the one from my previous message)

I would suggest disabling native symbol stripping, which would hopefully make the stack trace in the crash report a bit more descriptive:

```xml
<PropertyGroup>
    <NoSymbolStrip>true</NoSymbolStrip>
</PropertyGroup>
```

> * Cancel button does nothing, have to kill the process

I believe this has been fixed recently.

**Comment by jaosnz-rep** [comment]
This issue can't be reproduced on the latest 17.11.1

**Comment by acrigney** [comment]
Thanks heaps guys, yes I upgraded the app to net 9 with VS2022 preview and
the latest preview packages for all. But I am waiting on my developer to
get back to me.

**Comment by MitchBomcanhao** [comment]
the issue here talks says that they're reporting it against maui 8.0.70 SR7, but what their proj file says is maui 8.0.7, which is a 6 months old version.

**Comment by Raghu072021** [comment]
@acrigney  This sounds funny but I was facing same issue this resolved for me by  in properties-> IOS-> build- > keep debug link-framework sdk only select this option  it worked for me

**Comment by jdarwood007** [comment]
While testing for workarounds on another issue, #25534, I tested setting the linker to Full
```
	  <MtouchLink>Full</MtouchLink>
```

I was then getting this error.  I have a repo linked in that issue.

**Comment by jfversluis** [comment]
From what I gather this is fixed in more recent versions or not really an issue to begin with but a configuration error? I'm closing this here for now. If still something is wrong please let me know or open something new. Thanks!
